Add test cases of sideways-lr/rl writing-mode for scrollIntoView

cathiechen opened the pull request:
Hi,
This patch adds test with sideways-lr/rl writing-mode or rtl direction for scrollIntoView.
PTAL, thanks!

cathiechen committed:
Add test case for sideways writing-mode

cathiechen committed:
Add test case for sideways writing-mode

cathiechen committed:
change including patch

cathiechen committed:
change including patch

MyidShin committed:
Rename traits and mojo directory in //ui/gfx/geometry

This CL renames mojo to mojom and *_struct_traits.* to
*_mojom_traits.* in //ui/gfx/geometry.

*_struct_traits.* => *_mojom_traits.*
ui/gfx/geometry/mojo => ui/gfx/geometry/mojom

Bug: 806965
Change-Id: I288f4019f88815ac92fc2f038676e7459ea01bd0
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1717541
Commit-Queue: Miyoung Shin <myid.shin@igalia.com>
Reviewed-by: Kent Tamura <tkent@chromium.org>
Reviewed-by: Xiyuan Xia <xiyuan@chromium.org>
Reviewed-by: Scott Violet <sky@chromium.org>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Cr-Commit-Position: refs/heads/master@{#681266}

fred-wang commented on css/cssom-view/scrollIntoView-sideways-lr-writing-mode-and-rtl-direction.html:
```diff
@@ -0,0 +1,106 @@
+</div>
+
+<script>
+// In sideways-lr mode and rtl direction, X corresponds to the block axisand is oriented rightward.
```
missing space between "axis" and "and"

fred-wang commented on css/cssom-view/scrollIntoView-sideways-lr-writing-mode.html:
```diff
@@ -0,0 +1,108 @@
+// So the beginning edges are the bottom and left edges and the ending edges are the top and right edges.
+
+// According to the spec, y be min(0, max(y, element padding edge height - element scrolling area height)).
+// So y is nonpositive and decreases upward..
```
nit: double period.

cathiechen committed:
typos

cathiechen committed:
typos

FHorschig committed:
Reland "[LargestContentfulPaint] Set startTime"

This reverts commit 6df4545104337f8087db2c39ebb2ff8f8b03ac89.

Reason for revert: Reverting the revert might "fix" the test now that the failures are expected after a new wpt roll:
https://crrev.com/219a669145c403c99e3d7f441abbaad0e5d23a90/third_party/blink/web_tests/platform/win7/virtual/scalefactor200withoutzoom/external/wpt/largest-contentful-paint/observe-after-untrusted-scroll-expected.txt

Original change's description:
> Revert "[LargestContentfulPaint] Set startTime"
>
> This reverts commit 58051e426031caabd8ae28b96a9be1f92f3cd913.
>
> Reason for revert: Most suspicious CL for failing paint-related test
> failures as mentioned in the linked bug.
>
> Bug: 987915
>
> Original change's description:
> > [LargestContentfulPaint] Set startTime
> >
> > Currently startTime is 0, which is a problem because getEntries()
> > methods all sort by startTime. To help with this problem, we set the
> > startTime to renderTime, if available, or to loadTime, if renderTime is
> > nt available.
> >
> > Bug: 965505
> > Change-Id: Ie70b69269f7ce4be0fdc94f33a44a84e062d3ba5
> > Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1719379
> > Reviewed-by: Yoav Weiss <yoavweiss@chromium.org>
> > Commit-Queue: Nicolás Peña Moreno <npm@chromium.org>
> > Cr-Commit-Position: refs/heads/master@{#681147}
>
> TBR=npm@chromium.org,yoavweiss@chromium.org
>
> Change-Id: Ie810b8a3ac88369ab73a66670865f462f590e1ec
> No-Presubmit: true
> No-Tree-Checks: true
> No-Try: true
> Bug: 965505
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1719351
> Reviewed-by: Friedrich [CET] <fhorschig@chromium.org>
> Commit-Queue: Friedrich [CET] <fhorschig@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#681224}

TBR=npm@chromium.org,fhorschig@chromium.org,yoavweiss@chromium.org

Change-Id: I02478f0c560c64268593c3a9df158b4e505f4667
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Bug: 987915, 965505
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1720811
Reviewed-by: Friedrich [CET] <fhorschig@chromium.org>
Commit-Queue: Friedrich [CET] <fhorschig@chromium.org>
Cr-Commit-Position: refs/heads/master@{#681269}

lilles committed:
Remove element from whitespace reattach set when detached.

An element is added to the set if a child is removed in order to force
an update of whitespace siblings. Removing a child doesn't cause a
style recalc or layout tree rebuild, but removing an inline element
between say two whitespace nodes makes one of the whitespace LayoutTexts
unnecessary. We keep this set until the next lifecycle update and marks
the elements in the set for layout tree rebuild to reach the whitespace
children. We skip these nodes if the layout object had been detached in
the meantime.

What happens here, however is that we didn't remove the elements from
the set when the elements were removed from the document. In this case
it is even adopted into a different document and gets a new layout
object in that document's layout tree.

When the original document gets its lifecycle update, we try to rebuild
the layout tree of the element which is now in a different document and
DCHECKs fail for style dirtiness.

Bug: 981384
Change-Id: I26fe3a8ac2c6b8994e6d4ad9593a8476422a0535
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1719345
Reviewed-by: Anders Hartvoll Ruud <andruud@chromium.org>
Commit-Queue: Rune Lillesveen <futhark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#681312}

riju committed:
[ImageCapture] Add pan/tilt constraint and wire in Linux/CrOS.

Pan and Tilt Constraints are part of the UVC spec.
For Linux/CrOS, we can use V4L2 controls like
(V4L2_CID_PAN_ABSOLUTE and V4L2_CID_TILT_ABSOLUTE).

Spec: https://github.com/w3c/mediacapture-image/pull/182

Test Page: https://riju.github.io/WebCamera/samples/panTilt/

Putting Pan/Tilt feature behind a flag:
chrome --enable-blink-features=MediaCapturePanTilt

Intent to Implement and Ship discussions:
https://groups.google.com/a/chromium.org/d/msg/blink-dev/j-Q08QgBipM/F3a5sau1BwAJ

Bug: 934063
Change-Id: I552c4c8be717c3b67c4d91f826a1f16850430fa4
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1508519
Reviewed-by: Guido Urdaneta <guidou@chromium.org>
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Reviewed-by: Miguel Casas <mcasas@chromium.org>
Reviewed-by: Reilly Grant <reillyg@chromium.org>
Commit-Queue: Rijubrata Bhaumik <rijubrata.bhaumik@intel.com>
Cr-Commit-Position: refs/heads/master@{#681332}

EiraGe committed:
Reland "Re-write test pointerevent_pointermove_in_pointerlock.html"

This reverts commit 02efc0f4be0797b2c151e4aa598b125a6eccf191.

Reason for revert: [Cherry-picked & WIP] reland with fixing the test.

Original change's description:
> Revert "Re-write test pointerevent_pointermove_in_pointerlock.html"
>
> This reverts commit 9b7bafff73e59dc5d0cfcd0f499aa20be66bca54.
>
> Reason for revert: Causing leak failures on WebKit Linux Leak bot
>
> https://luci-milo.appspot.com/p/chromium/builders/ci/WebKit%20Linux%20Leak/2348
>
> Original change's description:
> > Re-write test pointerevent_pointermove_in_pointerlock.html
> >
> > The test was failing in wpt because it passing non-integer value to
> > test driver. This CL changes it to use "origin" instead.
> > This CL rewrites the test to makes it cleaner, instead of adding event
> > listener inside event listener.
> > This CL also changes the test to send the "move" events after receiving
> > pointerlockchange.
> >
> > Bug: 986277
> > Change-Id: I0f51523c3c90791339877b2e51afa8e9a4cc9d4d
> > Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1717195
> > Reviewed-by: Lan Wei <lanwei@chromium.org>
> > Commit-Queue: Ella Ge <eirage@chromium.org>
> > Cr-Commit-Position: refs/heads/master@{#680563}
>
> TBR=lanwei@chromium.org,eirage@chromium.org
>
> Change-Id: Ia53d1066363ceaf804f61ada369a4126c859f875
> No-Presubmit: true
> No-Tree-Checks: true
> No-Try: true
> Bug: 986277
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1716038
> Reviewed-by: Noel Gordon <noel@chromium.org>
> Commit-Queue: Noel Gordon <noel@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#680743}

TBR=noel@chromium.org,lanwei@chromium.org,eirage@chromium.org

Bug: 986277
Change-Id: I9b104597d2965261104e5842855ab235bdec8081
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1719227
Reviewed-by: Ella Ge <eirage@chromium.org>
Reviewed-by: Noel Gordon <noel@chromium.org>
Commit-Queue: Ella Ge <eirage@chromium.org>
Cr-Commit-Position: refs/heads/master@{#681360}

Commit:
Add missing renderer-side check for navigator.cookieEnabled.

We weren't doing the renderer-side security checks for navigator.cookieEnabled,
so they would get rejections browser-side. I am not sure this is the real
cause of 983090, but it is a real bug.

This became an issue after the big port of document.cookie (and navigator.cookieEnabled)
to RestrictedCookieManager, as previously-used RenderFrameMessageFilter::OnCookiesEnabled
was missing browser-side checks, while the new RestrictedCookieManager::CookiesEnabledFor
has them.

Bug: 983090
Change-Id: I12bd582c952691276ed149fb44d1c4a07f564fdb
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1715601
Commit-Queue: Maks Orlovich <morlovich@chromium.org>
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Cr-Commit-Position: refs/heads/master@{#681364}

fred-wang committed:
Add tests to check mapping of CSS attributes. (#18111)

This also introduce a new helper to generate MathML fragments for each tag.

autofoolip committed:
Update interfaces/shape-detection-api.idl (#18106)

* Update interfaces/shape-detection-api.idl

Source: https://github.com/tidoust/reffy-reports/blob/ae0ae16/whatwg/idl/shape-detection-api.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/563873234

* Make idlharness test secure origin

* Run on all workers

* Fix setup failure-case

* Apply suggestions from code review

Revert unrelated change

frivoal committed:
[css-writing-modes] tests propagation of writing-mode and direction from body to root (#18114)

npm1 committed:
[LargestContentfulPaint] Add TAO tests

Bug: 965505
Change-Id: I1463f4b029e57f5293dc8dbe62462db7d241bf45
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1721022
Reviewed-by: Yoav Weiss <yoavweiss@chromium.org>
Commit-Queue: Nicolás Peña Moreno <npm@chromium.org>
Cr-Commit-Position: refs/heads/master@{#681464}

rwlbuis committed:
Fix border-001.html (#18113)

The border style also needs to be set before the widths have any effect.
To keep things non intrusive transparent is chosen for the color.

fred-wang committed:
Add padding/border/margin tests for more MathML elements. (#18121)

* Add padding/border/margin tests for more MathML elements.

This also introduces useful helpers in mathml-fragments.js

TalbotG committed:
Added 24 tests and 7 references to cover issue 3066 (#18122)

Added 24 tests and 7 references to cover issue https://github.com/w3c/csswg-drafts/issues/3066

cathiechen committed:
Merge branch 'sideways' of https://github.com/cathiechen/wpt into sideways
